R/utils.R

```diff
@@ -57,3 +58,41 @@ modify_vector <- function(x, y = NULL) {
   lnames <- function(x) tolower(names(x))
   c(x[!(lnames(x) %in% lnames(y))], y)
 }
+
+#' Generate Bearer auth token
+#'
+#' This function allows to generate new auth token base on client id and secret
+#' If you looking for more infromation see: \url{https://auth0.com/docs/api/management/v2/tokens}
+#' @param client_id The OAuth application id
+#' @param client_secret The OAuth application secret
+#' @param .domain By default \code{\link{api_domain}}
+#' @return Auth0 Bearer token
+#' @export
+generate_token <- function(client_id, client_secret, .domain = api_domain()) {
+  resp <-
+    httr::POST(
+      url = build_url(parse_url(glue("{.domain}/oauth/token"))),
+      body = jsonlite::toJSON(list(
+        client_id = client_id,
+        client_secret = client_secret,
+        audience = build_url(parse_url(glue("{.domain}/api/v2/"))),
+        grant_type = "client_credentials"
+      ), auto_unbox = TRUE),
+      httr::add_headers(default_send_headers)
+    )
+
+  if (httr::http_type(resp) != "application/json") {
+    stop("API did not return json", call. = FALSE)
+  }
+
+  parsed <- jsonlite::fromJSON(httr::content(resp, "text", encoding = "UTF-8"), simplifyVector = FALSE)
+
+  if (httr::http_error(resp)) {
+    stop(glue("Auth0 API request failed [{httr::status_code(resp)}]\n{parsed$error_description}"), call. = FALSE)
+  }
+
+  set_token(parsed$access_token)
+  message(glue("The new token was set in the environment."))
+  browser()
```

Comment:
browser() statement

Reply:
Thanks for catch this. That was just a debugged line that I forget to remove.